src/contracts/interfaces/IPaymentCoordinator.sol

```diff
@@ -0,0 +1,175 @@
+    struct StrategyAndMultiplier {
+        IStrategy strategy;
+        // weight used to compare shares in multiple strategies against one another
+        uint96 multiplier;
+    }
+
+    struct RangePayment {
+        // Strategies & relative weights of shares in the strategies
+        StrategyAndMultiplier[] strategiesAndMultlipliers;
+        IERC20 token;
+        uint256 amount;
+        uint64 startTimestamp;
+        uint64 duration;
+    }
+
+    struct DistributionRoot {
+        // timestamp after which the root can be claimed against
+        uint64 activatedAfter;
+        // merkle root of the distribution
+        bytes32 root;
+    }
+
+    struct ClaimsTreeMerkleLeaf {
+        IERC20 token;
+        uint256 amount;
+        // The staker or operator who earned these tokens
+        address earner;
+    }
+
+    struct PaymentMerkleClaim {
+        ClaimsTreeMerkleLeaf[] leaves;
+        // The index of the root in the list of roots
+        uint32 rootIndex;
+        // proof of the claimaint's account root in the Merkle tree
+        uint32 accountIndex;
+        bytes accountTreeProof;
+        // The indices and proofs of the leafs in the claimaint's merkle tree for this root
+        uint32[] leafIndices;
+        bytes[] tokenTreeProofs;
+    }
+
+    /// EVENTS ///
+
+    event RangePaymentCreated(address indexed avs, bytes32 rangePaymentHash, RangePayment rangePayment);
+    event RangePaymentForAllCreated(address indexed submitter, bytes32 rangePaymentHash, RangePayment rangePayment);
+    event PaymentUpdaterSet(address indexed oldPaymentUpdater, address indexed newPaymentUpdater);
+    event PayAllForRangeSubmitterSet(address indexed payAllForRangeSubmitter, bool indexed oldValue, bool indexed newValue);
+    event ActivationDelaySet(uint64 oldActivationDelay, uint64 newActivationDelay);
+    event CalculationIntervalSecondsSet(uint64 oldCalculationIntervalSeconds, uint64 newCalculationIntervalSeconds);
+    event GlobalCommissionBipsSet(uint16 oldGlobalCommissionBips, uint16 newGlobalCommissionBips);
+    event RecipientSet(address indexed account, address indexed recipient);
+    event RootSubmitted(
+        bytes32 root,
+        uint32 paymentCalculationStartTimestamp,
+        uint32 paymentCalculationEndTimestamp,
+        uint32 activatedAt
+    );
+    event PaymentClaimed(ClaimsTreeMerkleLeaf leaf);
+
+    /// VIEW FUNCTIONS ///
+
+    /// @notice The address of the entity that can update the contract with new merkle roots
+    function paymentUpdater() external view returns (address);
+
+    /**
+     * @notice The interval in seconds at which the calculation for range payment distribution is done.
+     * @dev Payment durations must be multiples of this interval.
+     */
+    function calculationIntervalSeconds() external view returns (uint64);
+
+    /// @notice The maximum amount of time that a range payment can end in the future
+    function MAX_PAYMENT_DURATION() external view returns (uint64);
```

Comment:
mmm ok. so this means that retroactivate payments might need to broken up into ranges of `MAX_PAYMENT_DURATION` until they cover the required retroactive range. that's fine, just something to keep track of